src/main/java/io/github/connorhartley/guardian/internal/checks/HorizontalSpeedCheck.java

```diff
@@ -111,120 +108,145 @@ public SequenceBlueprint getSequence() {
-                                        .type("horizontally overspeeding")
-                                        .severity(travelDisplacement - maximumSpeed);
+                                    if (user.getPlayer().get().get(Keys.VEHICLE).isPresent()) {
+                                        return new ConditionResult(false, report.build(false));
+                                    }
 
-                                // TODO : Remove this after testing \/
-                                plugin.getLogger().warn(user.getName() + " has triggered the horizontal speed check and overshot " +
-                                        "the maximum speed by " + (travelDisplacement - maximumSpeed) + ".");
-                            } else {
-                                return new ConditionResult(false, successReportBuilder.build(false));
-                            }
+                                    double travelDisplacement = Math.abs(Math.sqrt((
+                                            (present.getX() - start.getX()) *
+                                                    (present.getX() - start.getX())) +
+                                            (present.getZ() - start.getZ()) *
+                                                    (present.getZ() - start.getZ())));
 
-                            return new ConditionResult(true, successReportBuilder.build(true));
-                        }
+                                    travelDisplacement += playerControlModifier / 2;
 
-                        return new ConditionResult(false, sequenceResult);
-                    })
+                                    double maximumSpeed = playerControlSpeed * blockModifier * (((
+                                            ((1 / ((playerControlTicks + blockModifierTicks) / 2)) *
+                                                    ((long) this.analysisTime * 1000)) + (currentTime - lastAction)) / 2) / 1000) + 0.01;
+
+                                    report
+                                            .information("Horizontal travel speed should be less than " + maximumSpeed +
+                                                    " while they're " + playerControlState.name() + ".");
+
+                                    if (travelDisplacement > maximumSpeed) {
+                                        report
+                                                .information("Overshot maximum speed by " + (travelDisplacement - maximumSpeed) + ".")
+                                                .type("horizontally overspeeding")
+                                                .severity(travelDisplacement - maximumSpeed);
+
+                                        // TODO : Remove this after testing \/
+                                        plugin.getLogger().warn(user.getName() + " has triggered the horizontal speed check and overshot " +
```

Comment:
You utilize logging levels. Why not have a debug level so you don't have to remind yourself to remove these as they could be good to keep for future debugging purposes.